packages/next/next-server/server/next-server.ts

```diff
@@ -784,6 +784,12 @@ export default class Server {
     query: ParsedUrlQuery = {},
     parsedUrl?: UrlWithParsedQuery
   ): Promise<void> {
+    if (!pathname.startsWith('/')) {
+      throw new Error(
```

Comment:
Thinking about this more we'll probably have to warn instead of error otherwise it might break certain applications using a custom server. Ideally I'd want to throw here though.

Reply:
Let's throw in development and warn in production maybe 🤔

Reply:
Sounds good, I will agree with this change 👌